provider/scripts/spark/offline_store_spark_runner.py

```diff
@@ -365,13 +379,16 @@ def execute_sql_query(
                 f"the output format '{output_format}' is not supported. Supported types: 'parquet', 'csv'"
             )
 
-        try:
-            stats_directory = f"{output_uri.rstrip('/')}/stats"
-            stats_df = display_data_metrics(output_dataframe, spark)
-            stats_df.write.json(stats_directory, mode="overwrite")
-        except Exception as e:
-            print(e)
-            print("Failed to display data metrics")
+        if job_type == JobType.MATERIALIZATION:
+            try:
+                stats_directory = f"{output_uri.rstrip('/')}/stats"
+                stats_df = display_data_metrics(output_dataframe, spark)
+                stats_df.write.json(stats_directory, mode="overwrite")
+            except Exception as e:
```

Comment:
i know you didn't touch this but does it make sense to swallow all exceptions here?

Reply:
Yes. Based on the discussion earlier, we don't won't to fail the job if the metrics fail.